packages/core/src/state-management/repository.ts

```diff
@@ -50,12 +50,18 @@ export class Repository {
    */
   stateManager: StateManager;
 
-  constructor(limit: number, private readonly logger: DiagnosticsLogger) {
+  /**
+   * @param cacheLimit - Maximum number of documents to store in memory cache.
+   * @param logger - Where we put diagnostics messages.
+   * @param concurrencyLimit - Maximum number of concurrently running tasks on the documents.
+   */
+  constructor(cacheLimit: number, private readonly logger: DiagnosticsLogger, concurrencyLimit?: number) {
     this.loadingQ = new NamedTaskQueue((error) => {
       logger.err(error);
     });
-    this.executionQ = new ExecutionQueue(logger, (docId) => this.get(docId));
-    this.inmemory = new StateCache(limit, (state$) => state$.complete());
+    const effectiveConcurrencyLimit = cacheLimit || concurrencyLimit;
```

Comment:
Should this be the `min()`?  What if concurrencyLimit is less than cacheLimit?

Reply:
These two limits are orthogonal, really like dimensions. If concurrencyLimit is less than cacheLimit, perceived node performance would suffer, as only `concurrencyLimit` documents can be _processed_ concurrently. Other `cacheLimit - concurrencyLimit` documents would sit idle waiting for their turn.

Reply:
Yep, but in that case isn't that the behavior the user asked for?  What's the rationale for always choosing `cacheLimit` over `concurrencyLimit`?  Feels like that makes the `concurrencyLimit` param meaningless except when you have an infinite sized cache (I'm assuming that `cacheLimit` of 0 means "no limit"?  otherwise the `concurrenclyLimit` would literally never be used)

I could imagine a scenario where running with a lower concurrencyLimit than cacheLimit could be valuable if for example you really want to avoid going to pubsub by keeping lots of docs in cache, but don't want a large amount processing in parallel

Reply:
Actually, instead of `min` or `||`, should we just not be considering `cacheLimit` here at all and just always take the `concurrencyLimit` as given?  There's already logic at a higher level to default `concurrencyLimit` to `cacheLimit` if `concurrencyLimit` isn't explicitly specified.  So maybe if the user configures an explicit `concurrencyLimit` we should just use it, regardless of what the `cacheLimit` is?

Reply:
Holy cow. It should be `concurrencyLimit || cacheLimit`. And not even here, since this is set during Ceramic init process.